WordPress/gutenberg
# Heading level dropdown: remove obtrusive tooltips in favor of visible text

Fixes https://github.com/WordPress/gutenberg/issues/50402
See https://github.com/WordPress/gutenberg/issues/55927
See https://github.com/WordPress/gutenberg/pull/46003

## What?

Removes the dropdown menu heading level tooltips in favor of visible text.

## Why?

Since the dropdown menu has now a vertical orientation, the tooltips of the menu buttons hide the other buttons on hover and focus. This is a functional and usability / accessibility problem that needs to be fixed.

I'd also like to remind that icon-only buttons have inherent usability and accessibility problems. Their usage should be limited to only the cases where there isn't enough space to use visible text. This is not the case, as there's a lot of space to show visible text.

## How?

Makes the dropdown option buttons use both an icon and visible text.
Having both the icon and visible text seems to be the most appropriate solution here. The icon represents a visual hint that may help some users, while the visible text is universally accessible to everyone.

Other examples of dropdown menus that use both an icon and visible text are already present in the block toolbar, for example the block switcher menu and the More options menu.

This PR also adds a stylesheet to the component, which is only used to increase a little the horizontal gap between the heading icons and the visible text. This is not uncommon, as there are already components that override the button component default gap value, for example the Link control and the block switcher.

## Testing Instructions




- Edit a post and add a heading block.
- Click the 'Change level' button in the toolbar.
- Observe the dropdown menu shows both an icon and visible text.
- Observe there is no tooltip on hover and focus.
- Go to the Site editor.
- Select the Site title and click the 'Change level' button in the toolbar.
- Observe the dropdown menu shows both an icon and visible text.
- Observe the dropdown includes also the Paragraph menu item.
- Observe there is no tooltip on hover and focus.

### Testing Instructions for Keyboard


## Screenshots or screencast 

Before:

![before](https://github.com/WordPress/gutenberg/assets/1682452/e5364461-efd6-4246-8337-aa34a89f898b)

After:

![after](https://github.com/WordPress/gutenberg/assets/1682452/05be1015-5055-41be-943f-73977ba21e67)
